packages/ts-type-expand-plugin/src/service/vue.ts

```diff
@@ -0,0 +1,140 @@
+  __volar__?: { language: Language }
+  // https://github.com/vuejs/language-tools/blob/v2.0.16/packages/typescript-plugin/index.ts#L75
+  __vue__?: { language: Language }
+}
+
+let oldProgram: VueProgram | undefined
+
+export function getPositionOfLineAndCharacterForVue(
+  ctx: {
+    program: ts.Program
+    ts: __ts | undefined
+  },
+  fileName: string,
+  startPos = -1,
+) {
+  const compilerOptions = {
+    ...ctx.program.getCompilerOptions(),
+    rootDir: ctx.program.getCurrentDirectory(),
+    declaration: true,
+    emitDeclarationOnly: true,
+    allowNonTsExtensions: true,
+  }
+
+  oldProgram = oldProgram ?? ctx.program
+
+  if (!oldProgram.__vue__ && !oldProgram.__volar__) {
+    if (!ctx.ts) {
+      return startPos
+    }
+
+    const options: ts.CreateProgramOptions = {
+      host: ctx.ts.createCompilerHost(compilerOptions),
+      rootNames: ctx.program.getRootFileNames(),
+      options: compilerOptions,
+      oldProgram: oldProgram,
+    }
+
+    let vueOptions: VueCompilerOptions
+    const createProgram = proxyCreateProgram(
+      ctx.ts,
+      ctx.ts.createProgram,
+      (ts, _options) => {
+        const { configFilePath } = _options.options
+        vueOptions =
+          typeof configFilePath === 'string'
+            ? createParsedCommandLine(
+                ts,
+                ts.sys,
+                configFilePath.replace(windowsPathReg, '/'),
+              ).vueOptions
+            : resolveVueCompilerOptions({
+                extensions: ['.vue', '.cext'],
+              })
+        return [
+          createVueLanguagePlugin(
+            ts,
+            (id) => id,
+            _options.host?.useCaseSensitiveFileNames() ?? false,
+            () => '',
+            () =>
+              _options.rootNames.map((rootName) =>
+                rootName.replace(windowsPathReg, '/'),
+              ),
+            _options.options,
+            vueOptions,
+          ),
+        ]
+      },
+    )
+
+    console.log('create vue program')
+    oldProgram = createProgram(options) as VueProgram
+  }
+
+  const language = (oldProgram.__volar__ ?? oldProgram.__vue__)?.language
+  if (language?.scripts) {
+    const vFile = language.scripts.get(fileName)
+    if (vFile?.generated?.root && vFile.generated.root.languageId === 'vue') {
+      const code = vFile.generated.root.embeddedCodes?.[0]
+      if (code) {
+        const sourceMap = new SourceMaps.SourceMap(code.mappings)
+
+        const serviceScript =
+          vFile.generated.languagePlugin.typescript?.getServiceScript(
+            vFile.generated.root,
+          )
+        if (serviceScript) {
+          const map = language.maps.get(serviceScript.code, vFile.id)
+          if (map) {
+            for (const [generatedOffset, mapping] of toGeneratedOffsets(
+              vFile,
+              map,
+              startPos,
+            )) {
+              console.log(JSON.stringify({ generatedOffset, mapping }))
```

Comment:
want: Please use project defined logger for logging.

ref: https://github.com/d-kimuson/ts-type-expand/blob/76443e30e14d675b34947bbca1dd1fa58f64a106/packages/ts-type-expand-plugin/src/index.ts#L17-L20